src/modules/recorder.module.ts

```diff
@@ -0,0 +1,373 @@
+}
+
+async function getClipFromS3(clip: Clip): Promise<Buffer> {
+    const params = {
+        Bucket: BUCKET,
+        Key: `${clip.id}.pcm`,
+        Range: `bytes=${clip.clipstart}-${clip.clipend}`
+    }
+
+    return new Promise<Buffer>((resolve, reject) => {
+        s3.getObject(params, (err, data) => {
+            if (err) {
+                reject(err)
+            } else if (!(data.Body instanceof Buffer)) {
+                reject(data.Body)
+            } else {
+                resolve(data.Body)
+            }
+        })
+    })
+    
+}
+
+async function playClip(channel: VoiceChannel, clipName: string, textChannel: TextChannel) {
+    const voiceChannelData = voiceChannels.get(channel.id)
+    if (!voiceChannelData) {
+        return
+    }
+
+    let clip
+    try {
+        clip = await getClipByName(clipName)
+    } catch (err) {
+        return await textChannel.send("Could not find a clip with that name.")
+    }
+
+    getClipFromS3(clip).then(buffer => {
+        // data from s3 is stored in single channel. need to convert to stereo
+        const converted = Buffer.alloc(buffer.length * 2)
+        for (let i = 0; i < buffer.length; i += 2) {
+            converted[i * 2] = buffer[i]
+            converted[i * 2 + 1] = buffer[i + 1]
+            converted[i * 2 + 2] = buffer[i]
+            converted[i * 2 + 3] = buffer[i + 1]
+        }
+
+        const readable = new Readable({
+            read() {
+                readable.push(converted)
+                readable.push(null)
+            }
+        })
+
+        voiceChannelData.connection.play(readable, {type: "converted"})
+    }).catch(err => {
+        console.log(err)
+    })
+}
+
+async function handleListClips(textChannel: TextChannel) {
+    const clips = await getAllClips()
+    return await textChannel.send(clips.map(c => c.name).sort().join(", "))
+}
+
+async function handleRenameClip(oldName: string, newName: string, textChannel: TextChannel) {
+    getClipByName(oldName).then(async clip => {
+        renameClip(oldName, newName).then(async () => {
+            return await textChannel.send(`Successfully renamed clip ${oldName} to ${newName}.`)
+        }).catch(async err => {
+            return await textChannel.send("Clip with that name already exists.")
+        })
+    }).catch(async err => {
+        return await textChannel.send("Could not find a clip with that name.")
+    })
+}
+
+async function handleDeleteClip(clipName: string, requester: string, textChannel: TextChannel) {
+    let clip
+    try {
+        clip = await getClipByName(clipName)
+    } catch (err) {
+        return await textChannel.send("Could not find a clip with that name.")
+    }
+
+    if (!clip.participants.includes(requester)) {
+        return await textChannel.send("You were not in this clip and cannot delete it.")
+    }
+
+    const params = {
+        Bucket: BUCKET,
+        Key: `${clip.id}.pcm`
+    }
+
+    s3.deleteObject(params, async (err, data) => {

```

Comment:
I see this is being awaited on line 336. `handleDeleteClip` resolves the Promise **before** the object is deleted. I'm not sure if that is intentional or not.